thirdparty/cmdconfig/commands/cmdeval/cmdeval.go

```diff
@@ -361,19 +361,46 @@ func checkFnConfigPathExistence(path string) error {
 	return nil
 }
 
-func (r *EvalFnRunner) preRunE(c *cobra.Command, args []string) error {
+func (r *EvalFnRunner) validateOptionalFlags() error {
 	// Let users know that --include-meta-resources is no longer necessary
 	// since meta resources are included by default.
 	if r.IncludeMetaResources {
 		return fmt.Errorf("--include-meta-resources is no longer necessary because meta resources are now included by default")
 	}
+	// SaveFn stores function to Kptfile. If not enabled, only make in-place changes.
+	if r.SaveFn {
+		if r.FnType == "" {
+			return fmt.Errorf("--type is required if saving functions to Kptfile (--save=true)")
```

Comment:
nit: try to be consistent with the eror style guide https://github.com/GoogleContainerTools/kpt/blob/main/docs/style-guides/errors.md

An alternate language:

function type (validator or mutator) must be specified using `--type`

Reply:
Wow, didn't know we even have an error guide. Updated